packages/clerk-js/src/ui/components/SessionTasks/tasks/TaskResetPassword/__tests__/TaskResetPassword.test.tsx

```diff
@@ -0,0 +1,145 @@
+import userEvent from '@testing-library/user-event';
+import { describe, expect, it } from 'vitest';
+
+import { bindCreateFixtures } from '@/test/create-fixtures';
+import { render, waitFor } from '@/test/utils';
+
+import { TaskResetPassword } from '..';
+
+const { createFixtures } = bindCreateFixtures('TaskResetPassword');
+
+describe('TaskResetPassword', () => {
+  it('does not render component without existing session task', async () => {
+    const { wrapper } = await createFixtures(f => {
+      f.withUser({
+        email_addresses: ['test@clerk.com'],
+        identifier: 'test@clerk.com',
+      });
+    });
+
+    const { queryByText, queryByRole } = render(<TaskResetPassword />, { wrapper });
+
+    expect(queryByText('New password')).not.toBeInTheDocument();
+    expect(queryByText('Confirm password')).not.toBeInTheDocument();
+    expect(queryByText('Sign out of all other devices')).not.toBeInTheDocument();
+    expect(queryByRole('link', { name: /sign out/i })).not.toBeInTheDocument();
+  });
+
+  it('renders component when session task exists', async () => {
+    const { wrapper } = await createFixtures(f => {
+      f.withUser({
+        email_addresses: ['test@clerk.com'],
+        identifier: 'test@clerk.com',
+        tasks: [{ key: 'reset-password' }],
+      });
+    });
+
+    const { queryByText, queryByRole } = render(<TaskResetPassword />, { wrapper });
+
+    expect(queryByText('New password')).toBeInTheDocument();
+    expect(queryByText('Confirm password')).toBeInTheDocument();
+    expect(queryByText('Sign out of all other devices')).toBeInTheDocument();
+    expect(queryByRole('link', { name: /sign out/i })).toBeInTheDocument();
+  });
+
+  it('renders the task components in the exact order of the tasks returned by the API', async () => {
+    const { wrapper } = await createFixtures(f => {
+      f.withOrganizations();
+      f.withForceOrganizationSelection();
+      f.withUser({
+        email_addresses: ['test@clerk.com'],
+        identifier: 'test@clerk.com',
+        tasks: [{ key: 'reset-password' }, { key: 'choose-organization' }],
+      });
+    });
+
+    const { queryByText, queryByRole } = render(<TaskResetPassword />, { wrapper });
+
+    expect(queryByText('New password')).toBeInTheDocument();
+    expect(queryByText('Confirm password')).toBeInTheDocument();
+    expect(queryByText('Sign out of all other devices')).toBeInTheDocument();
+    expect(queryByRole('link', { name: /sign out/i })).toBeInTheDocument();
+  });
```

Comment:
I believe the assertions of this test aren't matching it's description. It'd only make sense if it was coming from within `SignIn` or `SignUp` + triggering navigation with `setActive`.